src/main/java/org/apache/sling/feature/cpconverter/handlers/IndexDefinitionsEntryHandler.java

```diff
@@ -126,8 +126,9 @@ public void handle(@NotNull String path, @NotNull Archive archive, @NotNull Entr
 
                 } else {
                     // binary file, should we attach?
-                    if ( archive.getMetaInf().getFilter().contains(repositoryPath)) {
-                        indexManager.getIndexes().registerBinary(repositoryPath, is);
+                    String binaryParentPath  = repositoryPath.substring(0,repositoryPath.lastIndexOf("/")) ;
+                    if ( archive.getMetaInf().getFilter().contains(binaryParentPath)) {
+                        indexManager.getIndexes().registerBinary(binaryParentPath, is);
```

Comment:
for me it would make more sense to find the parent within the `registerBinary` method.

Reply:
The javadoc says explicitly:

> repositoryPath The JCR repository path where the binary was found

This is for me the path to the binary node (nt:file)